### DISCUSSION TITLE
Process to contract projects relying on deprecated formula?

### DISCUSSION THREAD

**Discussion by johnalanwoods** [Homebrew/discussions#4973] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.25
ORIGIN: https://github.com/Homebrew/brew
HEAD: c32bd1c7cc0c3b1b914845bddfeda53f4d877a3f
Last commit: 4 days ago
Core tap JSON: 11 Dec 15:22 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.43.0 => /opt/homebrew/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.6.2-arm64
CLT: N/A
Xcode: 15.0.1
Rosetta 2: false
```
```


### Output of `brew doctor`

```shell
: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
```

### Description of issue

I noticed that a number of serious projects, including LaTeX, pstoedit etc, still have some type of dependency on openssl@1.1 -

Is there a formal process to contact these projects on the Homebrew side? - does this deprecation ultimately end up with broken formula? 

This has been the case for a couple of months now.

**Comment by SMillerDev** [comment, marked as answer]
> Is there a formal process to contact these projects on the Homebrew side? - does this deprecation ultimately end up with broken formula?

No, Homebrew does not set deprecation dates for OpenSSL, OpenSSL upstream does. So it's up to individual projects to be ready for that.

**Reply by johnalanwoods** [reply]
So if projects don't respond they will forever be dependent on something that's deprecated, potentially breaking them?

**Reply by johnalanwoods** [reply]
ps. It's hard to even help projects, as it's a little confusing where the dependency lies, for example, 

```brew uses --installed openssl@1.1 --recursive``` yields:

```thefuck```


but `brew deps thefuck --tree` shows:

```
thefuck
├── python@3.12
│   ├── mpdecimal
│   ├── openssl@3
│   │   └── ca-certificates
│   ├── sqlite
│   │   └── readline
│   └── xz
└── six
```

so I don't really get it, how can we find out where the formula that needs the update lies?

**Reply by SMillerDev** [reply]
What does `brew outdated` say? Because nothing in core still depends on openssl@1.1.

**Reply by johnalanwoods** [reply]
@SMillerDev - thanks for all the help (as usual).

`brew outdated` shows nothing.
The full output of `brew uses --installed openssl@1.1` is:
`liblqr`
`pstoedit`
`thefuck`


maybe even more weirdly (assuming I understand these command's intention:

`brew uses --eval-all openssl@1.1` is:

```
ansible@2.9             appscale-tools          crackpkcs               erlang@22               glib-openssl            mariadb@10.2            mariadb@10.4            minimal-racket          mysql@5.7               python@3.7              sphinx                  sslyze
ansible@6               arangodb                erlang@21               erlang@23               libvnc                  mariadb@10.3            mariadb@10.7            mysql-client@5.7        node@14                 ruby@2.6                ssh-permit-a38
```

weird, as `--eval-all` is supposed to include installed and all available.

**Reply by johnalanwoods** [reply]
@SMillerDev - updated details above, any ideas on why I'm seeing this?

**Reply by SMillerDev** [reply]
`brew uses --eval-all` evaluates the state of the formula definitions. `brew uses --installed` evaluates the state of the installed formulae. `liblqr`, `pstoedit` and `thefuck` were rebuild to remove a reference to `openssl@1.1` but no update was forced on users since it does not affect the usage of the tools. If you reinstall them the usage should disappear.

**Reply by johnalanwoods** [reply]
@SMillerDev you are the best dude! This makes perfect sense, thank you for helping me understand this :)
